apps/src/code-studio/components/progress/teacherPanel/SelectedStudentInfo.jsx

```diff
@@ -65,6 +66,31 @@ export default class SelectedStudentInfo extends React.Component {
     }
   };
 
+  renderPartners({partners, partnerCount}) {
+    // Three cases:
+    // - no known names: "3 other students"
```

Comment:
how do students get into a situation where they are pairing with someone anonymous?

Reply:
If a user or their progress (i.e. user_level row) is deleted, we aren't able to get the name of the partner.  I've described this a little bit [here](https://github.com/code-dot-org/code-dot-org/blob/8648289b537f3695c16f56c76d618a0812086cfb/dashboard/app/models/user_level.rb#L125-L128) where the data is retrieved, but perhaps a more detailed comment here would be helpful?

Also, would it be clearer if I called these partners "unknown partners" instead of "anonymous partners"?

Reply:
Ah, I see. A think a little comment that names are no longer known if past partners are deleted would've clearer things up for me; "unknown" and "anonymous" seem similar to me - up to you.